google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/RetryOnNewChannelConfiguration.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner.spi.v1;
+
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.rpc.StatusCode;
+import com.google.cloud.spanner.v1.stub.SpannerStubSettings;
+import org.threeten.bp.Duration;
+
+/** Configuration for RPCs that use a custom retry strategy of retrying on a new gRPC channel. */
+class RetryOnNewChannelConfiguration {
+  private final Boolean retryBeginTransactionOnNewChannel =
+      Boolean.parseBoolean(
+          System.getProperty(
+              "com.google.cloud.spanner.retry_begin_transaction_on_new_channel", "false"));
+  private final int retryDelayBeginTransactionOnNewChannel =
+      getIntProperty("com.google.cloud.spanner.retry_delay_begin_transaction_on_new_channel", 5);
+  // This is the maximum number of times that the custom retry logic will attempt to execute the
+  // RPC.
+  private final int maxAttemptsBeginTransactionOnNewChannel =
+      getIntProperty("com.google.cloud.spanner.max_attempts_begin_transaction_on_new_channel", 4);
+
+  // This is the number of times that Gax will retry the RPC on a given channel when retrying on a
```

Comment:
Given its an experimental feature do we really want to expose settings to set retry configuration? Can't we start just by exposing a default setting (of low duration). Wouldn't `com.google.cloud.spanner.retry_begin_transaction_on_new_channel` be enough?

We anyway don't foresee system properties as the long-term way of passing such attributes.